internal/clientconn/conninfo/conn_info.go

```diff
@@ -53,6 +58,30 @@ func (connInfo *ConnInfo) SetAuth(username, password string) {
 	connInfo.password = password
 }
 
+// Cursor returns the cursor value stored.
+// We use "db.collection" as the key to get the cursor.
+func (connInfo *ConnInfo) Cursor(_ uint64) iterator.Interface[int, any] {
+	connInfo.curRW.RLock()
+	defer connInfo.curRW.RUnlock()
+
+	return connInfo.cursor
```

Comment:
It will panic if the cursor is not set, do we expect such behavior?

Reply:
It will return `nil`, we test that behaviour in the unit test.